### PR TITLE
handling csv creation

### PR DESCRIPTION
**Pull request by vahana**
it now respects the order of _field attribute
it also requires to specify all fields and wont expend any nested objects automatically.

### CODE REVIEW

File: prf/utils/utils.py [hunk 1]
```diff
@@ -362,63 +362,74 @@ def qs2dict(qs):
     return dictset(parse_qsl(qs))
 
 
-class TabRenderer(object):
-    def __init__(self, info):
-        pass
+def dict2tab(data, query_fields=None, headers=None, fmt='csv', auto=False):
```
Comment (cyrilrbt) [inline prf/utils/utils.py]: I'd move this to a `prf.utils.csv` or something

Comment (cyrilrbt) [inline prf/utils/utils.py]: Reanem `fmt` to `format`

File: prf/utils/utils.py [hunk 2]
```diff
@@ -362,63 +362,74 @@ def qs2dict(qs):
...
+    query_fields = query_fields or []
+    headers = headers or []
 
-        def pop_(each, key):
-            val = each.pop(key, '')
-            if isinstance(val, (datetime, date)):
-                return val.strftime('%Y-%m-%dT%H:%M:%SZ')  # iso
-            else:
-                return unicode(val)
+    def pop_(each, key):
```
Comment (cyrilrbt) [inline prf/utils/utils.py]: Rename `pop_` to `_pop`

File: prf/utils/utils.py [hunk 3]
```diff
@@ -362,63 +362,74 @@ def qs2dict(qs):
...
+                    row.append(pop_(each, col))
 
             tabdata.append(row)
 
-        return tabdata.get_csv(quoting=tablib.compat.csv.QUOTE_MINIMAL)
+        return getattr(tabdata, fmt)
 
-    def __call__(self, value, system):
+    except:
+        import sys
```
Comment (cyrilrbt) [inline prf/utils/utils.py]: Move imports to top of file

File: prf/view.py [hunk 4]
```diff
@@ -124,7 +124,8 @@ def set_renderer(self):
         elif 'text/plain' in self.request.accept:
             self.request.override_renderer = 'string'
 
-        elif 'text/csv' in self.request.accept:
+        elif 'text/csv' in self.request.accept or\
```
Comment (cyrilrbt) [inline prf/view.py]: `pep8` recommends parentheses to break up lines

File: prf/utils/utils.py [hunk 5]
```diff
@@ -362,63 +362,74 @@ def qs2dict(qs):
...
-                return unicode(val)
+    def pop_(each, key):
+        val = each.pop(key, '')
+        if isinstance(val, (datetime, date)):
+            return val.strftime('%Y-%m-%dT%H:%M:%SZ')  # iso
+        else:
+            return unicode(val)
+
+    if auto and data:
+        query_fields = data[0].flat(keep_lists=0).keys()
```
Comment (cyrilrbt) [inline prf/utils/utils.py]: You might be overriding `query_fields`. I think you should raise an error if `auto` and `query_fields` are passed.

File: prf/utils/utils.py (hunk 1 above)
Comment (cyrilrbt) [inline prf/utils/utils.py]: `headers` and `auto` don't need to be params, they can just be variables.

Comment (cyrilrbt) [inline prf/utils/utils.py]: Missing unit tests for this method

Comment (vahana) [inline prf/utils/utils.py]: removed headers and auto
renamed query_fields to fields and fmt to format_ (format is a python keyword)
added tests

File: prf/utils/utils.py (hunk 2 above)
Comment (vahana) [inline prf/utils/utils.py]: done

File: prf/utils/utils.py (hunk 5 above)
Comment (vahana) [inline prf/utils/utils.py]: removed auto logic all together

**Pull request merged**